src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardRepository.java

```diff
@@ -786,6 +900,22 @@ public void restore() throws IOException {
                     throw new IndexShardRestoreFailedException(shardId, "Failed to fetch index version after copying it over", e);
                 }
                 recoveryState.getIndex().updateVersion(segmentCommitInfos.getVersion());
+
+                /// now, go over and clean files that are in the store, but were not in the snapshot
+                try {
+                    for (String storeFile : store.directory().listAll()) {
+                        if (!Store.isChecksum(storeFile) && !snapshotFiles.containPhysicalIndexFile(storeFile)) {
+                            try {
+                                store.deleteQuiet("restore", storeFile);
+                                store.directory().deleteFile(storeFile);
+                            } catch (IOException e) {
+                                // ignore
+                            }
+                        }
+                    }
+                } catch (IOException e) {
+                    // ignore
```

Comment:
We shouldn't ignore this exception right? This could mean that the Directory couldn't be listed, and as a result none of the files that should be removed on restore are being removed. Should we at least warn here?